staging/src/k8s.io/pod-security-admission/admission/admission_test.go

```diff
@@ -365,8 +377,28 @@ func TestValidateNamespace(t *testing.T) {
 		},
 
 		// TODO: test for aggregating pods with identical warnings
-		// TODO: test for bounding evalution time with a warning
-		// TODO: test for bounding pod count with a warning
+		{
+			name:           "bounding evaluation time exceeded with a warning during update to restricted",
+			newLabels:      map[string]string{api.EnforceLevelLabel: string(api.LevelRestricted)},
+			oldLabels:      map[string]string{},
+			expectAllowed:  true,
+			expectListPods: true,
+			listDuration:   2 * time.Second,
+			expectEvaluate: api.LevelVersion{Level: api.LevelRestricted, Version: api.LatestVersion()},
+			expectWarnings: []string{"noruntimeclasspod: message", "Timeout reached after checking 1 pods"},
```

Comment:
this is a surprising warning message... if our list timed out, I wouldn't have expected us to be able to check any pods